src/lib.rs

```diff
@@ -334,28 +334,9 @@ fn replace_last(s: &mut String, n: usize, replacement: &str) {
 }
 
 fn replace_end_if(haystack: &mut String, search_for: &str, replacement: &str) {
-    if haystack.len() >= search_for.len() {
-        let start = haystack.len() - search_for.len();
-        replace_if(
-            haystack,
-            |h| h.ends_with(search_for),
-            start,
-            search_for.len(),
-            replacement,
-        );
-    }
-}
-
-fn replace_if<F: Fn(&str) -> bool>(
-    haystack: &mut String,
-    search_fn: F,
-    start_at: usize,
-    len: usize,
-    replacement: &str,
-) {
-    // if haystack[start_at..].starts_with(search_for) {
-    if search_fn(&haystack[start_at..]) {
-        haystack.replace_range(start_at..(start_at + len), replacement);
+    if haystack.strip_suffix(search_for).is_some() {
+        let start_at = haystack.len() - search_for.len();
+        haystack.replace_range(start_at..(start_at + search_for.len()), replacement);
```

Comment:
Some of that extra code was because of weird behavior if the search_for length is greater than the haystack. I'm not sure if our test set includes one and two char words but id not, we should definitely add them to make sure this works okay still.  If it does, nice job simplifying and fixing!  When I was doing my own testing outside of unit tests, I would feed in my local dictionary and run the binary tool to go over the whole thing to make sure it didn't get upset on any words. That might be worth doing with this change. I would also look to see how many duplicates and what it thought were duplicates, but I don't think you need to do that.

Reply:
I added various tests and haven't been able to break it. I also depended on it locally and used it to index 10,000 wikipedia documents and didn't run into any issues.

Reply:
Awesome, thanks! Feel free to merge.